skdownscale/pointwise_models/quantile.py

```diff
@@ -397,6 +401,122 @@ def _more_tags(self):
+        X = check_array(X, ensure_2d=True)
+
+        X = X[:, 0]
+
+        sort_inds = np.argsort(X)
+
+        X_cdf = self._calc_extrapolated_cdf(X[sort_inds], sort=False, extrapolate=self.extrapolate)
+        X_train_vals = np.interp(x=X_cdf.pp, xp=self._X_cdf.pp, fp=self._X_cdf.vals)
+
+        # generate y value as historical y plus/multiply by quantile difference
+        if self.kind == 'difference':
+            diff = X_cdf.vals - X_train_vals
+            sorted_y_hat = np.interp(x=X_cdf.pp, xp=self._y_cdf.pp, fp=self._y_cdf.vals) + diff
+        elif self.kind == 'ratio':
+            ratio = X_cdf.vals / X_train_vals
+            print(ratio)
+            if self.max_ratio is not None:
+                ratio = np.min(ratio, self.max_ratio)
+            sorted_y_hat = np.interp(x=X_cdf.pp, xp=self._y_cdf.pp, fp=self._y_cdf.vals) * ratio
+
+        # put things into the right order
+        y_hat = np.full_like(X, np.nan)
+        y_hat[sort_inds] = sorted_y_hat[1:-1]
+
+        # If extrapolate is 1to1, apply the offset between X and y to the
+        # tails of y_hat
+        if self.extrapolate == '1to1':
+            X_fit_len = len(self._X_cdf.vals)
+            X_fit_min = self._X_cdf.vals[0]
+            X_fit_max = self._X_cdf.vals[-1]
+
+            y_fit_len = len(self._y_cdf.vals)
+            y_fit_min = self._y_cdf.vals[0]
+            y_fit_max = self._y_cdf.vals[-1]
+
+            # adjust values over fit max
+            inds = X > X_fit_max
+            if inds.any():
+                if X_fit_len == y_fit_len:
+                    y_hat[inds] = y_fit_max + (X[inds] - X_fit_max)
+                elif X_fit_len > y_fit_len:
+                    X_fit_at_y_fit_max = np.interp(
+                        self._y_cdf.pp[-1], self._X_cdf.pp, self._X_cdf.vals
+                    )
+                    y_hat[inds] = y_fit_max + (X[inds] - X_fit_at_y_fit_max)
+                elif X_fit_len < y_fit_len:
+                    y_fit_at_X_fit_max = np.interp(
+                        self._X_cdf.pp[-1], self._y_cdf.pp, self._y_cdf.vals
+                    )
+                    y_hat[inds] = y_fit_at_X_fit_max + (X[inds] - X_fit_max)
+
+            # adjust values under fit min
+            inds = X < X_fit_min
+            if inds.any():
+                if X_fit_len == y_fit_len:
+                    y_hat[inds] = y_fit_min + (X[inds] - X_fit_min)
+                elif X_fit_len > y_fit_len:
+                    X_fit_at_y_fit_min = np.interp(
+                        self._y_cdf.pp[0], self._X_cdf.pp, self._X_cdf.vals
+                    )
+                    y_hat[inds] = X_fit_min + (X[inds] - X_fit_at_y_fit_min)
+                elif X_fit_len < y_fit_len:
+                    y_fit_at_X_fit_min = np.interp(
+                        self._X_cdf.pp[0], self._y_cdf.pp, self._y_cdf.vals
+                    )
+                    y_hat[inds] = y_fit_at_X_fit_min + (X[inds] - X_fit_min)
```

Comment:
How different is this from the base QuantileMappingRegressor implementation? Would it make sense to move this logic to a shared `_extrapolate_1_to_1` method?